### PR TITLE
New User Model & Item/Eatery Favoriting

### PR DESCRIPTION
**Pull request by cassidyxu**
## Overview

This PR completes the user model implementation and adds endpoints for managing user favorites, allowing users to add, view, and remove their favorite eateries and items.

## Changes Made

- **User Model**:
    - Created user model with necessary fields and relationships.
    - Added attributes to support the management of user-specific data like favorites.
- **Endpoints for Favorites**:
    - Added endpoints in `views.py` to handle adding, accessing, and removing user favorite items/eateries.
    - Updated the serializers to manage user favorite data.

## Next Steps

Explore solutions to extend user session duration beyond Cornell's default logout timeframe, ensuring users remain logged in to manage their favorites without interruption.

### CODE REVIEW

File: src/user/views.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    serializer_class = UserSerializer
+
+    @action(detail=True, methods=["post"], url_path="eatery/add")
+    def add_favorite_eatery(self, request, pk=None):
+        user = get_object_or_404(User, pk=pk)
+        eatery_id = request.data.get("eatery_id")
+        eatery = get_object_or_404(Eatery, id=eatery_id)
+        user.favorite_eateries.add(eatery)
+        user.save()
+        return Response({"status": "eatery added"}, status=status.HTTP_200_OK)
```
Comment (Aayush-Agnihotri) [inline src/user/views.py]: I think we can change the responses to maybe instead return the user back to frontend with the update eatery field

File: src/user/views.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        return Response({"status": "item added"}, status=status.HTTP_200_OK)
+
+    @action(detail=True, methods=["post"], url_path="item/remove")
+    def remove_favorite_item(self, request, pk=None):
+        user = get_object_or_404(User, pk=pk)
+        item_id = request.data.get("item_id")
+        item = get_object_or_404(Item, id=item_id)
+        user.favorite_items.remove(item)
+        user.save()
+        return Response({"status": "item removed"}, status=status.HTTP_200_OK)
```
Comment (Aayush-Agnihotri) [inline src/user/views.py]: same with all the responses

**Review by Aayush-Agnihotri — APPROVED** [review]
LGTM

File: src/eatery_blue_backend/urls.py [hunk 3]
```diff
@@ -21,6 +21,5 @@
     path("item/", include("item.urls")),
     path("category/", include("category.urls")),
     path("report/", include("report.urls")),
-    path("person/", include("person.urls")),
-    path("docs/", schema_view.with_ui("swagger", cache_timeout=0)),
```
Comment (Aayush-Agnihotri) [inline src/eatery_blue_backend/urls.py:25]: Add back the swagger doc endpoint

File: .gitignore [hunk 4]
```diff
@@ -4,6 +4,7 @@
 .envrc
 .envlocal
 .envremote
+secrets/eatery-blue-f810e-firebase-adminsdk-ds3co-7902015462.json
```
Comment (Aayush-Agnihotri) [inline .gitignore:7]: can you send me this file

Comment (cassidyxu) [reply .gitignore:7]: i dont have that file

File: src/user/views.py (hunk 1 above)
Comment (cassidyxu) [reply src/user/views.py]: do you mean something like

```python
user_data = UserSerializer(user).data
return Response(user_data, status=status.HTTP_200_OK)

**Review by tjvignos — APPROVED** [review]
lgtm

**Pull request merged**